### ISSUE TITLE
[WinError 1455] The paging file is too small for this operation to complete.

### ISSUE THREAD

**Issue by HessamSH** [ultralytics/ultralytics#776]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Training

### Bug

`Exception has occurred: OSError       (note: full exception trace is shown but execution is paused at: <module>)
[WinError 1455] The paging file is too small for this operation to complete. Error loading "C:\Users\Omen\anaconda3\envs\yolov8\lib\site-packages\torch\lib\shm.dll" or one of its dependencies.`

### Environment

- YOLO: Ultralytics 8.0.25 
- Python: 3.9
- Torch: (conda) conda install pytorch==1.12.1 torchvision==0.13.1 torchaudio==0.12.1 cudatoolkit=11.6 -c pytorch -c conda-forge
- GPU: GEFORCE 1070

### Minimal Reproducible Example

```
from ultralytics import YOLO

def tr():
    # Load a model
    model = YOLO("yolov8n.yaml")  # build a new model from scratch

    # Train the model
    results = model.train(data=r"fish.yaml", epochs=50, workers=8)

if __name__=="__main__":
    tr()
```

### Additional

This is a problem I enountered on a different system too when training on GPU. And also on the current system, I have tried installing ultralytics with pip and also with conda. In all the cases I have encountered the same problem.
Here is the exact error:
`Exception has occurred: OSError       (note: full exception trace is shown but execution is paused at: <module>)
[WinError 1455] The paging file is too small for this operation to complete. Error loading "C:\Users\Omen\anaconda3\envs\yolov8\lib\site-packages\torch\lib\shm.dll" or one of its dependencies.`

Here is also the code that generates this issue:
```
from ultralytics import YOLO

def tr():
    # Load a model
    model = YOLO("yolov8n.yaml")  # build a new model from scratch

    # Train the model
    results = model.train(data=r"fish.yaml", epochs=50, workers=8)

if __name__=="__main__":
    tr()
```

Here is the YAML file for training:
```
train: D:\Hesam\yolov8\DATA\train\images
val: D:\Hesam\yolov8\DATA\test\images

nc: 1
names: ['Fish']
```

Additionally, the code works fine as long as I specify 'cpu' as device for training. Even though torch recognizes my GPU and returns true to cuda_is_available, when specifying GPU for training I encouted this issue.

I have installed ultralytics inside conda with pip,
deleted the installed torch (1.13 that doesn't support GPU) and installed cuda torch with this command:
`conda install pytorch==1.12.1 torchvision==0.13.1 torchaudio==0.12.1 cudatoolkit=11.6 -c pytorch -c conda-forge
`

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by HessamSH** [comment]
It seems that the problem concerns the number of workers. When I set it to 2 or 1 it works fine, but anything more than that just generates the error. 

This works fine:
` results = model.train(data=r"fish.yaml", epochs=50, workers=2)`

**Comment by sstainba** [comment]
a 1070 is a very old gpu.  it probably doesn't have the resources to handle this.  sounds like more of a hardware/environment issue than one with ultralytics.

**Comment by HessamSH** [comment]
> a 1070 is a very old gpu. it probably doesn't have the resources to handle this. sounds like more of a hardware/environment issue than one with ultralytics.

The problem was all the same on 3090 . It is irrelevant to the GPU apparently.

**Comment by sstainba** [comment]
> > a 1070 is a very old gpu. it probably doesn't have the resources to handle this. sounds like more of a hardware/environment issue than one with ultralytics.
> 
> The problem was all the same on 3090 . It is irrelevant to the GPU apparently.

It must also have to do with your dataset then.  I train on a 3060 and always use the default which is 8 loaders and have never seen this issue.

**Comment by HessamSH** [comment]
> > > a 1070 is a very old gpu. it probably doesn't have the resources to handle this. sounds like more of a hardware/environment issue than one with ultralytics.
> > 
> > 
> > The problem was all the same on 3090 . It is irrelevant to the GPU apparently.
> 
> It must also have to do with your dataset then. I train on a 3060 and always use the default which is 8 loaders and have never seen this issue.

It COULD be that my images are too large to have 8 workers although I very much doubt it since I got the same error on a totally different dataset, but the error handling is awful in this case. I could have absolutely no idea the problem is with workers argument, especially when it was set by default to 8 and raised an error. 
There isn't also any warning in the docs that this particular argument could raise this error.

**Comment by sstainba** [comment]
> > > > a 1070 is a very old gpu. it probably doesn't have the resources to handle this. sounds like more of a hardware/environment issue than one with ultralytics.
> > > 
> > > 
> > > The problem was all the same on 3090 . It is irrelevant to the GPU apparently.
> > 
> > 
> > It must also have to do with your dataset then. I train on a 3060 and always use the default which is 8 loaders and have never seen this issue.
> 
> It COULD be that my images are too large to have 8 workers although I very much doubt it since I got the same error on a totally different dataset, but the error handling is awful in this case. I could have absolutely no idea the problem is with workers argument, especially when it was set by default to 8 and raised an error. There isn't also any warning in the docs that this particular argument could raise this error.

Yeah the error messages in general aren't great.  I've run into several where the message and the fix didn't seem related in the least.  When my images are large I just end up running out of gpu memory so I have to reduce the batch size.  The auto calculation for batch size doesn't seem terribly reliable.  I've started reading my images to 800x800 for most of my stuff.

**Comment by bit-scientist** [comment]
[This ](https://github.com/ultralytics/yolov3/issues/1643#issuecomment-1406242529)may help

**Comment by pderrenger** [comment]
@bit-scientist it sounds like the issue might be related to the number of workers used when training the model. Reducing the number of workers to 2 or 1 seems to solve the problem, while using 8 workers generates an error. This could be due to resource limitations when handling a larger number of workers, especially with larger image datasets, although it is unexpected that this issue was encountered even with a different dataset. It seems that there is no warning in the documentation about the potential for the workers argument to raise this error.

Regarding the error messages, it's acknowledged that there may be room for improvement. Different issues and fixes might not always seem related, and the automatic batch size calculation might not always be reliable.

Additionally, you've mentioned trying running the images at a smaller size to avoid running out of GPU memory, which is a good troubleshooting approach in case of large images causing memory issues.

Lastly, you suggested a link that may provide some helpful insights, which is appreciated. Your feedback and experiences are valuable for improving the overall user experience, and your contribution to the community is appreciated!

**Comment by KY-wwang** [comment]
> > a 1070 is a very old gpu. it probably doesn't have the resources to handle this. sounds like more of a hardware/environment issue than one with ultralytics.
> 
> The problem was all the same on 3090 . It is irrelevant to the GPU apparently.

I also encountered this problem, may have something to do with virtual memory: pytorch inside some cuda related dll files are very large, and as long as you imported pytorch related packages, will go to load these files, and in the opening of the multi-process, each process will go to load these files, in fact, may not be used, so in windows use **virtual memory**. 

I tried to reallocate 1000000M to my virtual memory, problem solved.

**Comment by glenn-jocher** [comment]
@KY-wwang it's insightful to hear that the issue persisted across different GPUs and that increasing the virtual memory allocation resolved the problem. This suggests that the error may indeed be related to system resources rather than the specific hardware capabilities of the GPU. Thank you for sharing your solution; it could be beneficial for others experiencing similar issues. Adjusting virtual memory settings is a useful troubleshooting step when encountering resource-related errors during multi-process operations.
